Sources/Purchasing/StoreKit1/PaymentQueueWrapper.swift

```diff
@@ -77,8 +77,14 @@ class PaymentQueueWrapper: NSObject, PaymentQueueWrapperType {
         super.init()
     }
 
-    func finishTransaction(_ transaction: SKPaymentTransaction) {
+    func finishTransaction(_ transaction: SKPaymentTransaction, completion: @escaping () -> Void) {
+        // See `StoreKit1Wrapper.finishTransaction(:completion:)`.
+        // Technically this is a race condition, because `SKPaymentQueue.finishTransaction` is asynchronous
+        // In practice this method won't be used, because this class is only used in SK2 mode,
+        // and those transactions are finished through `SK2StoreTransaction`.
```

Comment:
Should we log an error if this is reached then? Also, not sure if we should finish transaction in this case, if we consider it an error... But I guess we should propagate the error somehow in that case.

Reply:
having to have this here doesn't really say wonders about our class design, we might wanna start thinking about alternatives soon

Reply:
Yeah for sure. This is needed because we avoid using `StoreKit1Wrapper` when SK2 is enabled. But turned out we did need a payment queue wrapper for some things.
And the fact that `finishTransaction` needs a reference to _some_ wrapper, because SK1 transactions are finished through the queue, but SK2’s are finished directly with the transaction :/

So yeah I agree this is a smell, but due to those 2 limitations I can’t think of a way to avoid it.